simulator/infra/memory/memory.cpp

```diff
@@ -44,12 +44,21 @@ FuncMemory::FuncMemory( uint32 addr_bits,
     memory.resize(set_cnt);
 }
 
-void FuncMemory::memcpy_host_to_guest( Addr dst, const Byte* src, size_t size)
+size_t FuncMemory::memcpy_host_to_guest( Addr dst, const Byte* src, size_t size)
 {
-    for ( size_t offset = 0; offset < size; ++offset)
-        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic) Low level access
-        alloc_and_write_byte( dst + offset, *(src + offset));
-}    
+    size_t offset = 0;
```

Comment:
In case of exception, we do not return anything.
It looks like we may return `size` and put iterating variable into the loop.

Reply:
May I add noexcept variants of these function as I did it before?